src/plugins/config/CountAwareSamplerConfig.ts

```diff
@@ -1,13 +1,18 @@
 import BasePluginConfig from './BasePluginConfig';
+import Utils from '../utils/Utils';
+import { envVariableKeys } from '../../Constants';
 const koalas = require('koalas');
 
 class CountAwareSamplerConfig extends BasePluginConfig {
```

Comment:
Regardsless from `CountAwareSamplerConfig` and `TimeAwareSamplerConfig`, If I talk about `MetricSamplerConfig`, there should be sampling results should be `OR` with eachother (should not override themselves).

For the following code snippet from the `MetricSamplerConfig`:
```
        if (this.countAwareSampler) {
            isSampled = this.countAwareSampler.isSampled();
        }

        if (this.timeAwareSampler) {
            isSampled = this.timeAwareSampler.isSampled();
        }

        if (this.customSampler) {
            if (typeof this.customSampler === 'function' && this.customSampler().isSampled) {
                isSampled = this.customSampler().isSampled();
            }
        }
```

It should something like that:
```
       if (this.countAwareSampler) {
            isSampled = isSampled || this.countAwareSampler.isSampled();
        }

        if (this.timeAwareSampler) {
            isSampled = isSampled || this.timeAwareSampler.isSampled();
        }

        if (this.customSampler) {
            if (typeof this.customSampler === 'function' && this.customSampler().isSampled) {
                isSampled = isSampled || this.customSampler().isSampled();
            }
        }
```

In here, we are calling each sampler even though one  of them returns `true`. The reason is that they might require to be invoked at every check (for ex. count based sampler)

Reply:
Also, I think similar changes should be applied to `TraceSamplerConfig` also

Reply:
I applied the OR logic to  both `MetricSamplerConfig ` and `TraceSamplerConfig `